cpp/src/arrow/flight/sql/client.h

```diff
@@ -329,6 +329,25 @@ class ARROW_FLIGHT_SQL_EXPORT FlightSqlClient {
   /// \param[in] info         The FlightInfo of the query to cancel.
   ::arrow::Result<CancelResult> CancelQuery(const FlightCallOptions& options,
                                             const FlightInfo& info);
+   
+  /// \brief Sets session options.
+  ///
+  /// \param[in] options            RPC-layer hints for this call.
+  /// \param[in] session_options    The session options to set.
+  ::arrow::Result<std::vector<SetSessionOptionResult>> SetSessionOptions(
+      const FlightCallOptions& options,
+      const std::vector<SessionOption>& session_options);
+
+  /// \brief Gets current session options.
+  ///
+  /// \param[in] options            RPC-layer hints for this call.
+  ::arrow::Result<std::vector<SessionOption>> GetSessionOptions(
```

Comment:
Should probably work (server implementation-defined) and should effectively be a no-op unless one of the option names is defined by the server to be set-once-and-don't-allow-changing in which case it might complain.